File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/AccountInfoDataCc.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AccountInfoDataCc._

case class AccountInfoDataCc (
  cc: Option[String],
cc_exp: Option[String],
name: Option[String],
address: Option[String],
city: Option[String],
state: Option[String],
zip: Option[String],
country: Option[String],
maxmind_riskscore: Option[String],
maxmind: Option[AccountInfoMaxMindResponse],
verified: Option[Boolean])

object AccountInfoDataCc {
  import DateTimeCodecs._

  implicit val AccountInfoDataCcCodecJson: CodecJson[AccountInfoDataCc] = CodecJson.derive[AccountInfoDataCc]
  implicit val AccountInfoDataCcDecoder: EntityDecoder[AccountInfoDataCc] = jsonOf[AccountInfoDataCc]
  implicit val AccountInfoDataCcEncoder: EntityEncoder[AccountInfoDataCc] = jsonEncoderOf[AccountInfoDataCc]
}

File: OPENAPI-REST-API/openapi-client/zapier/models/AccountInfoDataCc.js
const utils = require('../utils/utils');
const AccountInfoMaxMindResponse = require('../models/AccountInfoMaxMindResponse');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}cc`,
                label: `[${labelPrefix}cc]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}cc_exp`,
                label: `[${labelPrefix}cc_exp]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}name`,
                label: `[${labelPrefix}name]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}address`,
                label: `[${labelPrefix}address]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}city`,
                label: `[${labelPrefix}city]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}state`,
                label: `[${labelPrefix}state]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}zip`,
                label: `[${labelPrefix}zip]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}country`,
                label: `[${labelPrefix}country]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}maxmind_riskscore`,
                label: `[${labelPrefix}maxmind_riskscore]`,
                type: 'string',
            },
            ...AccountInfoMaxMindResponse.fields(`${keyPrefix}maxmind`, isInput),
            {
                key: `${keyPrefix}verified`,
                label: `[${labelPrefix}verified]`,
                type: 'boolean',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'cc': bundle.inputData?.[`${keyPrefix}cc`],
            'cc_exp': bundle.inputData?.[`${keyPrefix}cc_exp`],
            'name': bundle.inputData?.[`${keyPrefix}name`],
            'address': bundle.inputData?.[`${keyPrefix}address`],
            'city': bundle.inputData?.[`${keyPrefix}city`],
            'state': bundle.inputData?.[`${keyPrefix}state`],
            'zip': bundle.inputData?.[`${keyPrefix}zip`],
            'country': bundle.inputData?.[`${keyPrefix}country`],
            'maxmind_riskscore': bundle.inputData?.[`${keyPrefix}maxmind_riskscore`],
            'maxmind': utils.removeIfEmpty(AccountInfoMaxMindResponse.mapping(bundle, `${keyPrefix}maxmind`)),
            'verified': bundle.inputData?.[`${keyPrefix}verified`],
        }
    },
}
